src/Cask/Cask.cs

```diff
@@ -287,6 +205,108 @@ public static CaskKey GenerateKey(string providerSignature,
         return CaskKey.Encode(key);
     }
 
+
+    private static Range ComputeSignatureCharRange(SecretSize secretSize)
+    {
+        int secretSizeInBytes = (int)secretSize * 16;
+        int paddedSecretSizeInBytes = RoundUpTo3ByteAlignment(secretSizeInBytes);
+        int sensitiveDataCharOffset = paddedSecretSizeInBytes / 3 * 4;
+        return sensitiveDataCharOffset..(sensitiveDataCharOffset + 4);
+    }
+
+    private static Range ComputeSignatureByteRange(SecretSize secretSize)
+    {
+        int secretSizeInBytes = (int)secretSize * 16;
+        int paddedSecretSizeInBytes = RoundUpTo3ByteAlignment(secretSizeInBytes);
+        return paddedSecretSizeInBytes..(paddedSecretSizeInBytes + 3);
+    }
+
+    private static SecretSize InferSensitiveDataSizeFromCharLength(int lengthInChars)
+    {
+        Debug.Assert(IsValidKeyLengthInChars(lengthInChars));
+
+        int lengthInBytes = lengthInChars / 4 * 3;
+        return InferSensitiveDataSizeFromByteLength(lengthInBytes);
+    }
+
+    private static SecretSize InferSensitiveDataSizeFromByteLength(int lengthInBytes)
```

Comment:
nit: rename here too?
```suggestion
    private static SecretSize InferSecretSizeFromByteLength(int lengthInBytes)
```